lightning/src/ln/channelmonitor.rs

```diff
@@ -1915,12 +1932,15 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 		if let Some(ref mut cur_local_tx) = self.current_local_signed_commitment_tx {
 			if should_broadcast {
 				cur_local_tx.tx.add_local_sig(&self.key_storage.funding_key, self.funding_redeemscript.as_ref().unwrap(), self.channel_value_satoshis.unwrap(), &self.secp_ctx);
+				//TODO: move key access behind KeysInterface inside OnchainTxHandler
+				// If would_broadcast_at_height determine this local tx should be broadcast, absolute_timelock is set to current_height,
+				// because it indicates that this tx confirmation is urgent. It's not going to change anything because we can't bump
+				// local_commitment before anchor_outputs, should be rethought afterwards
+				claimable_outpoints.push(ClaimRequest { absolute_timelock: height, aggregable: false, outpoint: BitcoinOutPoint { txid: self.key_storage.funding_info.as_ref().unwrap().0.txid.clone(), vout: self.key_storage.funding_info.as_ref().unwrap().0.index as u32 }, witness_data: InputMaterial::Funding { local_tx_remote_signed: cur_local_tx.tx.clone(), channel_value: self.channel_value_satoshis.unwrap() }});
```

Comment:
clone() ing the cur_local_tx wholesale here really sucks. a) you need to drop the add_local_sig line above, since we should now be signing + broadcasting this tx in onchain wholesale, and b) can we take() the cur_local_tx.tx and pass it to OnChainTxHandler? We should never be doing anything with it aside from sign, then broadcast, so can it happen in onchain?